lib/ui/index.js

```diff
@@ -194,11 +193,13 @@ UI.prototype.prompt = function(questions, callback) {
 
   // If no callback was provided, automatically return a promise
   if (callback) {
-    inquirer.prompt(questions, callback);
-  } else {
-    return new Promise(function(resolve) {
-      inquirer.prompt(questions, resolve);
+    inquirer.prompt(questions).then(function(result) {
+      callback(result);
+    }).catch(function() {
+      callback(null);
```

Comment:
this is essentially just mirroring the previous behavior, but I'm open to other ideas.